### ISSUE TITLE
Keys which may need to be accent quoted are not handled properly

### ISSUE THREAD

**Issue by daylinmorgan** [tanelso2/yanyl#5]
Great to see more work on simplifying object creation with `yaml`. 

I ran into an issue when defining a type with backticks. See below for a modified example from the tests.

```nim
import
    yanyl

type
    Owner = ref object of RootObj
        name: string
    Pet = ref object of RootObj
        name: string
        `type`: string
        owner: Owner

deriveYamls:
    Owner
    Pet

let sample = """
name: Garfield
type: cat
owner:
    name: J. Arbuckle
"""

let garf = ofYamlStr(sample, Pet)
doAssert garf.name == "Garfield"
doAssert garf.`type` == "cat"
doAssert garf.owner.name == "J. Arbuckle"
```

Which then raises: 

```sh
/home/daylin/.nimble/pkgs/yanyl-0.0.3/yanyl/reflection.nim(142, 5) Error: unhandled exception: do not know how to get name of nnkAccQuoted [ValueError]
```

Seems a small change to the [`getNameWithoutStar` ](https://github.com/tanelso2/yanyl/blob/8e02633464115a0cd31c8e98d570cf3f1fb914f9/src/yanyl/reflection.nim#L131) may be necessary to account for this case.

**Comment by tanelso2** [comment]
Thanks for pointing this out!

I will work on a fix soon.

**Comment by daylinmorgan** [comment]
Great! I can also submit a PR with https://github.com/daylinmorgan/yanyl/commit/3cd975e22f69de4d18419c4a81e511ea8feaacfa if you want.

**Comment by tanelso2** [comment]
Huh, I thought for sure this would require adding quotes in the codegen part.

Fixed if you upgrade to version `0.0.4` !

Let me know if you find any other bugs!
